frontend/src/components/Chat/DirectMessageSelect.tsx

```diff
@@ -0,0 +1,84 @@
+import {
+    Box, Button, Heading, Table,
+    Tbody,
+    Td,
+    Th,
+    Thead,
+    Tr,
+} from '@chakra-ui/react';
+import React, { useCallback, useState } from 'react';
+import { MessageType } from '../../classes/MessageChain';
+import Player from '../../classes/Player';
+import useCoveyAppState from '../../hooks/useCoveyAppState';
+import ChatContainer from './ChatContainer';
+
+interface DirectMessageSelectProps {
+    isViewingChatContainer: boolean;
+    setIsViewingChatContainer: (isViewingChatContainer: boolean) => void;
+}
+export default function DirectMessageSelect({
+    isViewingChatContainer,
+    setIsViewingChatContainer,
+}: DirectMessageSelectProps): JSX.Element {
+    const { myPlayerID, players, directMessageChains } = useCoveyAppState();
+    const [chosenDirectID, setDirectID] = useState<string>('');
+    const playersWithChats: Player[] = [];
+    const playersWithoutChats: Player[] = [];
+
+    players.forEach((player) => {
+        const directMessageId = [myPlayerID, player.id].sort().join(':');
+        if (directMessageChains[directMessageId]) {
+            playersWithChats.push(player);
+        }
+        else if (player.id !== myPlayerID) {
+            playersWithoutChats.push(player);
+        }
+    })
+
+    const handleChat = async (otherPlayerID: string) => {
+        const directMessageId = [myPlayerID, otherPlayerID].sort().join(':');
+        setDirectID(directMessageId);
+        setIsViewingChatContainer(true);
+    };
+
+    if (isViewingChatContainer) {
+        return <ChatContainer
+            directMessageID={chosenDirectID}
+            chainType={MessageType.DirectMessage}
+        />;
+    }
+    return (
+        <div>
```

Comment:
is the indentation right in this file? it looks a little off to me

Reply:
it's right in the file- nothing happened when I re-ran formatting